### PR TITLE
[ci-visibility] Update dotnet Testing Instructions

### PR DESCRIPTION
**Pull request by tonyredondo**
### What does this PR do?
Adds a new block for custom instrumentation on dotnet for ci visibility.

### Motivation
Exact steps must be followed to properly use custom instrumentation inside a ci visibility test suite.

### Preview

https://docs-staging.datadoghq.com/tony/ci-visibility-update-ci-library/continuous_integration/setup_tests/dotnet/#custom-instrumentation

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/setup_tests/dotnet.md [hunk 1]
```diff
@@ -126,10 +126,28 @@ If you are running tests in non-supported CI providers or with no `.git` folder,
...
 
+## Custom instrumentation
+
+Your setup for custom instrumentation depends on the `dd-trace` version and includes additional steps:
+
+<div class="alert alert-warning">
+  <strong>Note:</strong> If you are using custom instrumentation, you must keep the package versions (dd-trace and Datadog.Trace NuGet packages) in sync.
+</div>
+
+To use custom instrumentation in your .NET application:
```
Comment (drodriguezhdez) [inline content/en/continuous_integration/setup_tests/dotnet.md:137]:
```suggestion
To use the custom instrumentation in your .NET application:
```

File: content/en/continuous_integration/setup_tests/dotnet.md [hunk 2]
```diff
@@ -126,10 +126,28 @@ If you are running tests in non-supported CI providers or with no `.git` folder,
...
+
+1. Execute `dd-trace --version` to get the version of the tool.
+1. Add the `Datadog.Trace` [NuGet package][3] with the same version to your application.
+2. In your application code, access the global tracer through the `Datadog.Trace.Tracer.Instance` property to create new spans.
+
+For more information on adding spans and tags for custom instrumentation, see the [.NET Custom Instrumentation documentation][4].
```
Comment (drodriguezhdez) [inline content/en/continuous_integration/setup_tests/dotnet.md:143]:
```suggestion
For more information about how to add spans and tags for custom instrumentation, see the [.NET Custom Instrumentation documentation][4].
```

File: content/en/continuous_integration/setup_tests/dotnet.md (hunk 1 above)
Comment (tonyredondo) [reply content/en/continuous_integration/setup_tests/dotnet.md:137]: this paragraph is an adapted copy from: https://docs-staging.datadoghq.com/tony/ci-visibility-update-ci-library/tracing/setup_overview/setup/dotnet-core/?tab=windows#custom-instrumentation

File: content/en/continuous_integration/setup_tests/dotnet.md (hunk 2 above)
Comment (tonyredondo) [reply content/en/continuous_integration/setup_tests/dotnet.md:143]: this paragraph is an adapted copy from: https://docs-staging.datadoghq.com/tony/ci-visibility-update-ci-library/tracing/setup_overview/setup/dotnet-core/?tab=windows#custom-instrumentation

**Review by drodriguezhdez — APPROVED** [review]

File: content/en/continuous_integration/setup_tests/dotnet.md [hunk 3]
```diff
@@ -126,10 +126,28 @@ If you are running tests in non-supported CI providers or with no `.git` folder,
 : Commit committer date in ISO 8601 format.<br/>
 **Example**: `2021-03-12T16:00:28Z`
 
+## Custom instrumentation
+
+Your setup for custom instrumentation depends on the `dd-trace` version and includes additional steps:
+
+<div class="alert alert-warning">
+  <strong>Note:</strong> If you are using custom instrumentation, you must keep the package versions (dd-trace and Datadog.Trace NuGet packages) in sync.
```
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_tests/dotnet.md:134]:
```suggestion
  <strong>Note:</strong> If you are using custom instrumentation, you must keep the package versions (`dd-trace` and `Datadog.Trace` NuGet packages) in sync.
```

File: content/en/continuous_integration/setup_tests/dotnet.md [hunk 4]
```diff
@@ -126,10 +126,28 @@ If you are running tests in non-supported CI providers or with no `.git` folder,
 : Commit committer date in ISO 8601 format.<br/>
 **Example**: `2021-03-12T16:00:28Z`
 
+## Custom instrumentation
+
+Your setup for custom instrumentation depends on the `dd-trace` version and includes additional steps:
```
Comment (alai97) [inline content/en/continuous_integration/setup_tests/dotnet.md:131]:
```suggestion
```

File: content/en/continuous_integration/setup_tests/dotnet.md [hunk 5]
```diff
@@ -126,10 +126,28 @@ If you are running tests in non-supported CI providers or with no `.git` folder,
...
+
+<div class="alert alert-warning">
+  <strong>Note:</strong> If you are using custom instrumentation, you must keep the package versions (`dd-trace` and `Datadog.Trace` NuGet packages) in sync.
```
Comment (alai97) [inline content/en/continuous_integration/setup_tests/dotnet.md:134]:
```suggestion
  <strong>Note:</strong> Your custom instrumentation setup depends on the `dd-trace` version. To use the custom instrumentation, you must keep the package versions for `dd-trace` and `Datadog.Trace` NuGet packages in sync.
```

**Review by alai97 — APPROVED** [review]

**Pull request merged**